### ISSUE TITLE
Gradients problem

### ISSUE THREAD

**Issue by amlarraz** [bermanmaxim/LovaszSoftmax#28]
Hi, I've tried to use your loss implementation using pytorch 1.4 and the network do not learn anything.  This is because you've used Variable to define some tensors in the loss and in this Pytorch version the Variable has been deprecated as you can see in:
[https://pytorch.org/docs/stable/autograd.html#variable-deprecated](url). 

I've fixed this problem by adding `requires_grad=True` in all Variable definitions in the file:
[https://github.com/bermanmaxim/LovaszSoftmax/blob/master/pytorch/lovasz_losses.py](url)

**Comment by brooklyn1900** [comment]
Hi, my lovaszsoftmax loss is around 0.35 and never get lower for training and validation, and the test iou is low. Do you know why? thanks

**Comment by amlarraz** [comment]
But it got stuck at this value or it never decreases?

**Comment by mrluin** [comment]
Hello @brooklyn1900 
I have tried LovaszSoftmaxLoss on different datasets. In one of them, I have the same problem with you, loss was always swing around at 0.30 and got nan value later. But on the other dataset, everything was normal.

Did you solve your problem?
